pkg/controller/perconaservermongodbrestore/psmdb_restore_controller.go

```diff
@@ -401,19 +409,17 @@ func (r *ReconcilePerconaServerMongoDBRestore) resyncStorage(
 		profileName := naming.BackupSourceProfileName(cr)
 
 		_, err = pbmC.GetConfig(ctx)
-		if err == nil {
-			if err := pbmC.AddProfile(ctx, r.client, cluster, profileName, stg); err != nil {
-				return errors.Wrap(err, "add backup source as profile")
-			}
-
-			if err := pbmC.ResyncProfileAndWait(ctx, profileName); err != nil {
-				return errors.Wrap(err, "start profile resync")
-			}
-
-			return nil
+		if err != nil && !backup.IsErrNoDocuments(err) {
+			return errors.Wrap(err, "get config")
 		}
 
-		if backup.IsErrNoDocuments(err) {
+		// PBM (v2.10.0) does not resync the oplog metadata when using `ResyncProfileAndWait`.
+		// See `percona-backup-mongodb/pbm/resync/rsync.go`:
+		// - `Resync` is called when syncing the main storage (using the operator's `ResyncMainStorageAndWait` method). It syncs the "oplog, backup, and restore meta".
+		// - `SyncBackupList` is called when syncing the profile storage (using the operator's `ResyncProfileAndWait` method). It only syncs backup metadata.
+		//
+		// When we want to perform a restore using PITR, we must configure the profile as the main storage.
+		if cr.Spec.PITR != nil || backup.IsErrNoDocuments(err) {
```

Comment:
if the err is `nil` I think we have a potential nil dereference on `backup.IsErrNoDocuments(err)`

Given that

```
func IsErrNoDocuments(err error) bool {
	return errors.Is(err, mongo.ErrNoDocuments) || strings.Contains(err.Error(), "no documents in result")
}

```

Reply:
https://github.com/percona/percona-server-mongodb-operator/pull/2050/commits/37580c711a1498c51388b03194ca583110c05941